src/cast_core/test/SamplingLoader_test.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+
+#include <boost/exception/diagnostic_information.hpp>
+
+#include <yaml-cpp/yaml.h>
+
+#include <gennylib/context.hpp>
+#include <regex>
+#include <testlib/ActorHelper.hpp>
+#include <testlib/MongoTestFixture.hpp>
+#include <testlib/helpers.hpp>
+
+namespace genny {
+namespace {
+using namespace genny::testing;
+namespace bson_stream = bsoncxx::builder::stream;
+using namespace bsoncxx;
+
+TEST_CASE_METHOD(MongoTestFixture,
+                 "SamplingLoader - demo",
+                 "[standalone][single_node_replset][three_node_replset][sharded][SamplingLoader]") {
+
+    dropAllDatabases();
+    auto db = client.database("test");
+    auto collection = db.collection("sampling_loader_test");
+    // Seed some data.
+    collection.insert_many(std::vector{from_json("{\"x\": 0}"),
+                                       from_json("{\"x\": 1}"),
+                                       from_json("{\"x\": 2}"),
+                                       from_json("{\"x\": 3}"),
+                                       from_json("{\"x\": 4}")});
+
+    NodeSource nodes = NodeSource(R"(
+        SchemaVersion: 2018-07-01
+        Clients:
+          Default:
+            URI: )" + MongoTestFixture::connectionUri().to_string() +
+                                      R"(
+        Actors:
+        # In order to test something this random, we'll use a sample size equal to the collection
```

Comment:
My only concern is reprodusibility of failures when the randomness cannot be controlled. If you think that's not an issue for Genny's workloads using this actor, I'm fine with that.

Reply:
I think it could be an issue in the future, but I don't see it being any time soon. This actor is really kind of just out of laziness. If you wanted to generate documents that look like some other documents but are deterministic, you can do so using the more standard document generators like other tests do. For example, this one: https://github.com/mongodb/genny/blob/master/src/phases/execution/ClusteredCollection.yml. I personally haven't used it, but my expectation is that those would have a random seed - or else it would be a lot easier to bake one in there than it would be here.